cmd/yaman/image/image.go

```diff
@@ -0,0 +1,102 @@
+package image
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strings"
+
+	specs "github.com/opencontainers/image-spec/specs-go/v1"
+)
+
+type Image struct {
+	Hostname string
+	Name     string
+	Version  string
+	RootDir  string
+	manifest *specs.Manifest
+	config   *specs.Image
+}
+
+var imageNamePattern = regexp.MustCompile("^[a-z0-9]+([._-][a-z0-9]+)*(/[a-z0-9]+([._-][a-z0-9]+)*)*$")
+
+func New(rootDir, name string) (*Image, error) {
+	parts := strings.Split(name, "/")
+	if len(parts) != 3 {
+		return nil, fmt.Errorf("image name must be fully qualified")
+	}
+	hostName := parts[0]
+	userName := parts[1]
+
+	parts = strings.Split(parts[2], ":")
+	imageName := userName + "/" + parts[0]
+	if !isNameValid(imageName) {
+		return nil, fmt.Errorf("invalid image name")
+	}
+
+	imageVersion := "latest"
+	if len(parts) > 1 {
+		imageVersion = parts[1]
+	}
+	// TODO: check image version
```

Comment:
We might not need here. We can let the registry attempt to download the image and... fail if the version does not exist.